launcher/src/main/java/org/apache/spark/launcher/CommandBuilderUtils.java

```diff
@@ -336,4 +334,18 @@ static void addPermGenSizeOpt(List<String> cmd) {
     cmd.add("-XX:MaxPermSize=256m");
   }
 
+  /**
+   * Get the major version of the java.version string supplied.
+   */
+  static int javaMajorVersion(String javaVersion) {
+    String[] version = javaVersion.split("[+.\\-]+");
```

Comment:
But they won't since we only try and access the second element of the array if the first element is less than or equal to 1.